### PR TITLE
posix: move everything related to inode_array to inode_array.c

### PR DESCRIPTION
**Pull request by marcinslusarz**
... and simplify orphaned_inodes cleanup logic.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/160)

### CODE REVIEW

File: src/libpmemfile-posix/inode_array.c [hunk 1]
```diff
@@ -149,3 +149,82 @@ inode_array_unregister(PMEMfilepool *pfp,
...
+
+	while (cur) {
+		if (cur->used) {
+			uint32_t used = cur->used;
+			for (unsigned i = 0; used && i < NUMINODES_PER_ENTRY;
+					++i) {
+				if (TOID_IS_NULL(cur->inodes[i]))
+					continue;
+				inode_cb(pfp, cur->inodes[i]);
+				used--;
```
Comment (GBuella) [inline src/libpmemfile-posix/inode_array.c]: How come this routine decreases the `used` field? The `inode_cb` must be `inode_free`? Then there is no point in having an `inode_cb` argument, if that only be one thing, as it sort of is true right now.

**Comment by krzycz** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/160)*

File: src/libpmemfile-posix/inode_array.c (hunk 1 above)
Comment (marcinslusarz) [inline src/libpmemfile-posix/inode_array.c]: "Used" is a local variable.

Comment (GBuella) [inline src/libpmemfile-posix/inode_array.c]: Aaah, I see. I could have a different name.
Also, I think there is no need for the `if (cur->used)` branch, as that is checked inside the loop condition anyways. Thus the indentation can be simpler.

Comment (marcinslusarz) [inline src/libpmemfile-posix/inode_array.c]: Done.

**Comment by GBuella** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/160#-:-KlUuBSSdsQ9kfBL1YbT:bnfp4nl)*

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/160#-:-KlV74sImjkzj4YLX9fr:bnfp4nl)*

**Pull request merged**